src/components/editor-component.ts

```diff
@@ -85,75 +86,44 @@ export default function(
       ),
       html.div(
         {
-          id: 'text-edit-tools',
+          id: 'editor',
         },
         [
-          html.button(
-            {
-              disabled: true,
-            },
-            'B',
-          ),
-          html.button(
-            {
-              disabled: true,
-            },
-            'I',
-          ),
-          html.button(
-            {
-              disabled: true,
-            },
-            'U',
-          ),
-          html.select(
+          html.textarea(
             {
-              disabled: false,
-            },
-            [
-              html.option(
-                {
-                  disabled: false,
-                },
-                'Body',
-             ),
-             html.option(
-               {
-                 disabled: false,
-               },
-               'Heading 1',
-             ),
-             html.option(
-               {
-                 disabled: false,
-               },
-               'Heading 2',
-             ),
-             html.option(
-               {
-                 disabled: false,
-               },
-               'Heading 3',
-             ),
-            ],
-          ),
-          html.button(
-            {
-              disabled: true,
+              id: 'text-editor',
+              oncreate: (el: El) => {
+                const mirrorMarkOptions = {
+                  showToolbar: true,
+                }
+                const mirrorMarkEditor = (window as any).mirrorMark(el, mirrorMarkOptions)
+                mirrorMarkEditor.render()
+
+                // Get the CodeMirror (editor) object.
+                const codeMirrorEditor = mirrorMarkEditor.cm
+
+                // Set the onChange event to capture input data.
+                codeMirrorEditor.on('change', () => {
+                  const textEditor = (document as any).getElementById('text-editor')
```

Comment:
The oncreate [gives you the element already as the first argument](https://github.com/jorgebucaran/hyperapp#oncreate). You don't need to or want to do this.

Reply:
Fixed.